### PR TITLE
Resolving race condition in snapshots flush

### PR DESCRIPTION
**Pull request by rishigupta1599**
Current flushing logic wait till only the last dequeued element to be resolve the promise and as soon as that happens we take early exit. Now it can be the case that there are still the snapshots in the queue which did not got flushed. Now if queue is marked as end, we will finalize the build, but as snpashots queue still has some elements it will try to finalize those after build is finalized till the time queue is cleared.

### CODE REVIEW

**Review by ninadbstack — APPROVED** [review]
LGTM

File: packages/core/test/unit/queue.test.js [hunk 1]
```diff
@@ -401,24 +401,32 @@ describe('Unit / Tasks Queue', () => {
...
 
-    resolve2();
-    await expectAsync(p2).toBeResolved();
+  it('empty flush', async () => {
+    q.set({ concurrency: 1 });
+    await q.start();
+    q.readyState = 2;
+    expect(q.size).toBe(0);
+    let promise = generatePromise(q.flush());
+    await expectAsync(promise).toBePending();
```
Comment (itsjwala) [inline packages/core/test/unit/queue.test.js:428]: Update expectation with debug logs added

**Review by itsjwala — APPROVED** [review]

File: packages/core/test/unit/queue.test.js (hunk 1 above)
Comment (rishigupta1599) [reply packages/core/test/unit/queue.test.js:434]: Done

File: packages/cli-app/package.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
   },
   "publishConfig": {
     "access": "public",
-    "tag": "beta"
+    "tag": "alpha"
```
Comment (itsjwala) [inline packages/cli-app/package.json:12]: revert package.json changes before merging

**Review by itsjwala — APPROVED** [review]

**Pull request merged**